social.md

```diff
@@ -11,134 +11,235 @@ support:
+        },
+        ...
+        "status": "ENABLED",
+        ...
+    }
+    ...
+}
+```
+
+The conditions necessary to include the `authorizeUri` provider URL in the framework integration's `accountStores` login model are:
+
+* A Social Provider Directory must be mapped to the Application
+* webConfig.status = ENABLED
+* webConfig.login.enabled = true
+
+### Optional Query Parameters
+
+#### `redirect_uri`
+
+The `redirect_uri` query parameter indicates the "last leg" of the flow. It should be a fully qualified url and it *must* be in the list of `authorizedCallbackUris` from the `/applications/:appID` endpoint.
 
-* Create a callback handler for the provider's page-based redirect flow (see
-  next section)
+If this query parameter is not included on the `/authorize` URL, then the first element of the `authorizedCallbackUris` list will be used.
 
+#### `scope`
 
-## Implementing Page-Based Workflows
+A default `scope` for an individual social provider is automatically set on the backend. This can be overridden by providing the `scope` query parameter.
 
-The library must implement provider callbacks for all social provider
-directories that are mapped to the specified Stormpath application.  The
-callback URLs for each provider should take the form of:
+#### `state`
 
-> `<stormpath.web.social.[providerId].uri>`
+If the `state` parameter is included, it will be passed back in the final leg of the flow to your application via the `redirect_uri`.
 
-Where `providerId` is the property that is found in the `provider` object of
-the Stormpath directory, e.g. `google` or `facebook`.
+This parameter can be user like a CSRF token. That is, during `/authorize` URL construction, you can save the `state` value and then compare it to the `state` value that's passed back in the final leg of the flow.
 
-When the callback URL is requested with a GET request, the user is being
-redirected back to the server after authentication with the provider.  The
-callback handler must complete the following tasks:
+## `redirect_uri` response
 
-  * Parse the callback data from the provider to get the authorization code
-  * Perform the authorization code exchange, if needed. 
-  * Retrieve or create the account, by posting the provider data to Stormpath
-  * Create the OAuth2 token cookies for the user
-  * Redirect the user to `stormpath.web.login.nextUri`
+The last leg of the social interaction is to send a `302` redirect to the browser. The `Location` will be the `redirect_uri` originally set in the `/authorize` URL at the beginning of the flow. It will include a `jwtResponse` query parameter as well as a `state` query parameter (if it was included at the beginning of the flow).
 
-If any of these tasks fail, an error should be immediately rendered to the user
-and the next task should not be attempted.
+The JWT set as the `jwtResponse` has an `stt` header parameter of `assertion`. It should be processed in the usual way by the framework integration.
```

Comment:
If this assertion was generated by the client-api, it will be signed using the api key from the web configuration.